indico/web/client/js/custom_elements/ind_datepicker.js

```diff
@@ -0,0 +1,631 @@
+          $calendarButton.textContent = date.getDate();
+          $calendarButton.dataset.currentMonth = date.getMonth() === month;
+          $calendarButton.setAttribute(
+            'aria-label',
+            date.toLocaleDateString(indCalendar.locale, {
+              year: 'numeric',
+              month: 'long',
+              day: 'numeric',
+            })
+          );
+          $calendarButton.toggleAttribute(
+            'data-weekend',
+            indCalendar.weekInfo.weekend.includes(date.getDay() + 1)
+          );
+          $calendarButton.tabIndex = -1; // reset tabIndex
+          $calendarButton.value = toISODateString(date);
+
+          // Mark range
+          $calendarButton.disabled = date < rangeStart || date > rangeEnd;
+          $calendarButton.toggleAttribute('data-range-start', sameDate(date, markStart));
+          $calendarButton.toggleAttribute('data-range-end', sameDate(date, markEnd));
+          $calendarButton.toggleAttribute('data-range', date < markEnd && date > markStart);
+
+          // Mark selected
+          if ($calendarButton.value === indCalendar.value) {
+            $calendarButton.setAttribute('aria-selected', true);
+          } else {
+            $calendarButton.removeAttribute('aria-selected');
+          }
+        }
+
+        const $focusableButton =
+          listbox.querySelector('[aria-selected]') || listbox.firstElementChild;
+        $focusableButton.tabIndex = 0;
+      }
+    }
+
+    disconnectedCallback() {
+      this.cleanup();
+    }
+
+    static observedAttributes = ['open', 'value', 'range-start', 'range-end'];
+
+    attributeChangedCallback(name) {
+      switch (name) {
+        case 'open':
+          this.dispatchEvent(new Event('attrchange.open'));
+          break;
+        case 'value':
+        case 'range-start':
+        case 'range-end':
+          this.dispatchEvent(new Event('attrchange.value'));
+          break;
+      }
+    }
+  }
+);
+
+// Date functions
+
+function sameDate(a, b) {
+  return a?.getTime() === b?.getTime();
+}
+
+function toISODateString(date) {
+  // The 'en-ca' locale gives us the local date in ISO format,
+  // and it's not the same as as the actual ISO date
+  return date?.toLocaleDateString('en-ca');
+}
+
+function toOptionalDate(dateString) {
+  const date = new Date(dateString);
+  if (!date.getTime()) {
+    return;
+  }
+  date.setHours(0, 0, 0, 0);
+  return date;
+}
+
+function formatDate(formatString, date) {
+  const year = `${date.getFullYear()}`;
+  const month = `${date.getMonth() + 1}`;
+  const day = `${date.getDate()}`;
+
+  return formatString
+    .replace(/yy+/, function(s) {
+      if (s.length === 2) {
+        return year.slice(3);
+      }
+      return year;
+    })
+    .replace(/m{1,2}/, function(s) {
+      return month.padStart(s.length, '0');
+    })
+    .replace(/d{1,2}/, function(s) {
+      return day.padStart(s.length, '0');
+    });
+}
+
+// Parser
+
+function createDateParser(formatString) {
```

Comment:
What do you think about **always** allowing people to type a date in ISO format (`yyyy-mm-dd`) in addition to the chosen date format? Personally I think it'd be nice (since there are no other `yyyy-??-??` style formats we allow to select when configuring the regform field), but maybe that's just the nerd in my who wants to be able to use ISO format instead of whatever weird (`mm/dd/yyyy` 🙈) format someone chose...

If this is not super trivial to implement or you think it's strange, feel free to ignore that suggestion though!

Reply:
> If this is not super trivial to implement or you think it's strange, feel free to ignore that suggestion though!

Hmmm.... Well, I think we could simply try to parse the date using the parser, and if that fails, try to parse using the Date constructor. Consider it an invalid date only when both fail. Something like that. It shouldn't be too much work.

Reply:
Hah, learned something new today. The `Date` constructor can parse all kinds of human-readable date (amazing!), but ISO date will be treated as UTC, so that's about the only format that we *cannot* use for this case. 😂 Oh well.